client/src/pages/index.js

```diff
@@ -68,19 +74,15 @@ export class Index {
         });
       }
     }
-
-    this.reloadItems();
-    this.loadStatistics();
-    this.initialised = true;
   }
 
-  attached() {
+  attachItemListScrollObserver() {
     // this observer will always observe the last item in the item list
     // it checks if the observed element is within the viewport
     // if so, we load more items
-    this.itemListScrollObserver = new IntersectionObserver(entries => {
+    this.itemListScrollObserver = new IntersectionObserver(async entries => {
       if (entries[0].isIntersecting && this.moreItemsAvailable !== false) {
-        this.loadMore();
+        await this.loadMore();
```

Comment:
Imo this doesn't really change the behaviour if this is `async`. I would not do it, at least to me it's not clear why this should be async.

Do you think we should block loading if it's already loading? then this should be done in the `loadItems` function as this callback is most probably called again no matter what the return value of the callback is (in this case a promise)

Reply:
Ah I see. No I think it doesn't make sense. Removed in in -> https://github.com/nzzdev/Q-editor/pull/139/commits/5242970b7d9c76956212ec3aaf460c417c58a0b6